apps/frontend/src/app/pages/company/profile/profile.component.ts

```diff
@@ -54,27 +111,119 @@ export class ProfileComponent {
+      this.areaOfExpertise !== this.form.value.areaOfExpertise ||
+      this.profilePhotoToCompare !== this.form.value.profilePhoto ||
+      !this.areEqual(
+        this.originalWhatIsTheCompanyLookingFor,
+        this.form.value.whatIsTheCompanyLookingFor as Array<string>
+      )
+    )
+      this.validForm = true;
+    else this.validForm = false;
+  }
+
+  setInitialValues(profile: CompanyProfile | null) {
+    this.form.setValue({
+      about: profile?.about ?? '',
+      areaOfExpertise: profile?.areaOfExpertise ?? '',
+      companyName: profile?.name ?? '',
+      profilePhoto: profile?.companyPhoto?.url ?? '',
+      whatIsTheCompanyLookingFor: profile?.whatIsTheCompanyLookingFor ?? [],
+    });
+    this.about = profile?.about ?? '';
+    this.areaOfExpertise = profile?.areaOfExpertise ?? '';
+    this.whatIsTheCompanyLookingFor = profile?.whatIsTheCompanyLookingFor ?? [];
+    this.companyName = profile?.name ?? '';
+    this.profilePhoto = profile?.companyPhoto?.url ?? '';
+    this.profilePhotoToCompare = this.profilePhoto;
+    this.originalWhatIsTheCompanyLookingFor =
+      profile?.whatIsTheCompanyLookingFor
+        ? [...profile.whatIsTheCompanyLookingFor]
+        : [];
+  }
+
+  getDto() {
+    const updateDto: UpdateCompanyProfile = {};
+    if (this.profilePhotoToCompare !== this.form.value.profilePhoto)
+      updateDto.companyPhoto = this.profilePhotoFile;
+    if (this.about !== this.form.value.about)
+      updateDto.about = this.form.value.about as string;
+    if (this.areaOfExpertise !== this.form.value.areaOfExpertise)
+      updateDto.areaOfExpertise = this.form.value.areaOfExpertise as string;
+    if (
+      !this.areEqual(
+        this.originalWhatIsTheCompanyLookingFor,
+        this.form.value.whatIsTheCompanyLookingFor as Array<string>
+      )
+    )
+      updateDto.whatIsTheCompanyLookingFor = this.form.value
+        .whatIsTheCompanyLookingFor as Array<string>;
+    if (this.companyName !== this.form.value.companyName)
+      updateDto.name = this.form.value.companyName as string;
+    return updateDto;
+  }
+
+  updateProfile() {
+    this.isUpdating = true;
+    const updateDto = this.getDto();
+    console.log(updateDto);
+    this.userService.updateCompanyProfile(updateDto).subscribe({
+      next: (profile) => {
+        this.isUpdating = false;
+        this.editProfile = false;
+        this.setInitialValues(profile);
+        this.toastService.showToast({
+          message: 'Profile successfully updated',
+          type: 'success',
+        });
+      },
+      error: () => {
+        this.isUpdating = false;
+        this.toastService.showToast({
+          message: 'Error while updating profile',
+          type: 'error',
+        });
+      },
+    });
   }
 
   toggleEditProfile() {
     this.editProfile = !this.editProfile;
   }
 
-  removeAreaOfInterest(index: number) {
-    this.areaOfInterest.splice(index, 1);
+  removeWhatIsTheCompanyLookingFor(index: number) {
```

Comment:
When a remove the last string of the looking for, this function transforms it in a empty string:

![Captura de tela de 2024-04-16 14-23-18](https://github.com/Venture-Miner/wagademy/assets/83927877/746847b0-47cc-4d91-8bc4-db166178919a)

Steps to reproduce:

1) Remove the last string from the looking for
2) Click in edit profile